LoveMachine.KK/KoikatsuButtplugController.cs

```diff
@@ -241,6 +241,37 @@ protected override bool IsIdle(int girlIndex) => !supportedModes.Contains(flags.
             RunStrokerLoop(girlIndex, bone);
     }
 
+    internal class KoikatsuButtplugRotatorController : KoikatsuButtplugController
+    {
+        private static readonly HFlag.EMode[] supportedModes =
+        {
+            HFlag.EMode.houshi, HFlag.EMode.sonyu, HFlag.EMode.houshi3P, HFlag.EMode.sonyu3P,
+            HFlag.EMode.houshi3PMMF, HFlag.EMode.sonyu3PMMF
+        };
+
+        private static readonly List<string> supportedAnimations = new List<string>
+        {
+            "WLoop", "SLoop",
+            // anal
+            "A_WLoop", "A_SLoop", "A_OLoop",
+            // orgasm
+            "OLoop", "A_OLoop",
+        }.Union(orgasmAnimations).ToList();
+
+        protected override bool IsIdle(int girlIndex) => !supportedModes.Contains(flags.mode)
+                    || !supportedAnimations.Contains(flags.nowAnimStateName)
+                    || flags.speed < 1;
+
+        protected override bool IsOrgasming(int girlIndex) =>
+            orgasmAnimations.Contains(flags.nowAnimStateName);
+
+        protected override void HandleFondle(float y, int girlIndex, Bone bone, float timeSecs) =>
+            DoRotate(girlIndex, bone, true, timeSecs);

```

Comment:
This is mainly for female players and IMHO the only sensible way to implement it with rotators would be to set the rotator speed to `y` (which is a value between 0 and 1). Or if you don't want to implement it, you can just throw `NotImplementedException` here and remove the aibu rotator controller.
(It would absolutely make sense to implement it though, as some female toys have rotation, like the Nora).